pkg/scheduler/framework/cycle_state_test.go

```diff
@@ -31,44 +31,88 @@ func (f *fakeData) Clone() StateData {
 	return copy
 }
 
-func TestCycleStateClone(t *testing.T) {
-	var key StateKey = "key"
-	data1 := "value1"
-	data2 := "value2"
+var key StateKey = "key"
 
-	state := NewCycleState()
-	originalValue := &fakeData{
-		data: data1,
-	}
-	state.Write(key, originalValue)
-	stateCopy := state.Clone()
+// createCycleStateWithFakeData creates *CycleState with fakeData.
+// The given data is used in stored fakeData.
+func createCycleStateWithFakeData(data string, recordPluginMetrics bool) *CycleState {
+	c := NewCycleState()
+	c.Write(key, &fakeData{
+		data: data,
+	})
+	return c
+}
 
-	valueCopy, err := stateCopy.Read(key)
-	if err != nil {
-		t.Errorf("failed to read copied value: %v", err)
-	}
-	if v, ok := valueCopy.(*fakeData); ok && v.data != data1 {
-		t.Errorf("clone failed, got %q, expected %q", v.data, data1)
+func TestCycleStateClone(t *testing.T) {
+	tests := []struct {
+		name            string
+		data            string // data should be stored in CycleState.
+		state           *CycleState
+		wantClonedState *CycleState // clonedState is CycleState cloned by Clone().
+	}{
+		{
+			name:            "clone with recordPluginMetrics true",
+			data:            "data",
+			state:           createCycleStateWithFakeData("data", true),
+			wantClonedState: createCycleStateWithFakeData("data", true),
+		},
+		{
+			name:            "clone with recordPluginMetrics false",
+			data:            "data",
+			state:           createCycleStateWithFakeData("data", false),
+			wantClonedState: createCycleStateWithFakeData("data", false),
+		},
+		{
+			name:            "clone with nil CycleState",
+			state:           nil,
+			wantClonedState: nil,
+		},
 	}
 
-	originalValue.data = data2
-	original, err := state.Read(key)
-	if err != nil {
-		t.Errorf("failed to read original value: %v", err)
-	}
-	if v, ok := original.(*fakeData); ok && v.data != data2 {
-		t.Errorf("original value should change, got %q, expected %q", v.data, data2)
-	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			state := tt.state
+			stateCopy := state.Clone()
+			if tt.wantClonedState == nil {
```

Comment:
I feel the logic can be simplified by leveraging `cmp.Diff()`. Does https://gist.github.com/Huang-Wei/5f4bd967d3adf3d738531c09214bc307 look better?

Reply:
Looks nice. Thanks